src/main/python/ayab/firmware_flash.py

```diff
@@ -147,32 +150,17 @@ def execute_flash_command(self):
         # else
         tr_ = QCoreApplication.translate
         try:
-            p = Popen(command, stdout=PIPE, stderr=STDOUT, shell=True)
```

Comment:
On second thoughts I'm not satisfied with this implementation of timeout because it hands over control of the thread to the shell process and freezes the GUI.

Reply:
Do you intend to change this or is this PR ready to merge?

Reply:
Let's merge the PR and put a "please test" label on the issue. I'll get back to it at a later date.